routes/signMessage.ts

```diff
@@ -1,19 +1,96 @@
-import { signMessage } from '../lib/signatures'
-import { app } from '../server'
+import { ed25519 } from '@noble/curves/ed25519'
+import { blake3 } from '@noble/hashes/blake3'
+import { base64 } from '@scure/base'
+import { app } from '../app'
+import { messageDataToUint8Array } from '../buffers/buffers'
+import { kms } from '../clients/aws'
+import { authDb } from '../database/watcher'
+import { selectKeysQuery, selectSessionQuery } from '../lib/queries'
+import { signMessageWithKey } from '../lib/signatures'
+import { lucia } from '../lucia/auth'
 
-app.post('/sign', async (c) => {
+import type { Message } from '../utils/types'
+import { isMessage } from '../utils/types'
+
+app.post('/signMessage', async (c) => {
   try {
-    const { message } = await c.req.json()
+    const { sessionId, message } = await c.req.json()
+
+    if (!sessionId || !message) {
+      return c.json({ success: false, message: 'Missing parameters' }, 400)
+    }
 
-    if (!message) {
-      return c.json({ success: false, message: 'No message provided' }, 400)
+    if (!isMessage(message)) {
+      return c.json({ success: false, message: 'Invalid message format' }, 400)
     }
 
-    const signedMessage = signMessage(message)
+    const { session, user } = await lucia.validateSession(sessionId)
+
+    if (!session) {
+      return c.json({ success: false, message: 'Invalid session' }, 404)
+    }
+
+    const sessionResult = await authDb.query(selectSessionQuery, [sessionId])
+    const { userid } = sessionResult.rows[0]
+
+    if (userid.toString() !== message.messageData.rid) {
+      return c.json({ success: false, message: 'RID mismatch' }, 400)
+    }
+
+    const keysResult = await authDb.query(selectKeysQuery, [userid])
+
+    if (keysResult.rows.length === 0) {
+      return c.json({ success: false, message: 'Keys not found' }, 404)
+    }
+
+    const { encryptedprivatekey, publickey } = keysResult.rows[0]
+
+    console.log('HRE', publickey)
+
+    const computedHash = blake3(messageDataToUint8Array(message.messageData))
+
+    const computedHashBase64 = base64.encode(computedHash)
+    console.log({ computedHashBase64 })
+    console.log({ messagehash: message.hash })
+
+    if (computedHashBase64 !== message.hash) {
```

Comment:
the more i think of this, i feel like the hash should be the pure bytes. adding a base64 encoding step is confusing. 

although, making it a string does generally make it easier to pass around in other environments (like evm for example), even tho we could just wait to convert uint8 array into hex at the last second (since we dont need it atm)

what do u think?

Reply:
either way the hash/encode or wtv needs to be in one helper function. otherwise too confusing / annoying to use